src/components/backgroundselector/BackgroundSelectorDirective.js

```diff
@@ -0,0 +1,138 @@
+              {id: 'ch.swisstopo.pixelkarte-farbe', label: 'bg_pixel_color'},
+              {id: 'ch.swisstopo.pixelkarte-grau', label: 'bg_pixel_grey'},
+              {id: 'voidLayer', label: 'void_layer'}];
+
+          scope.backgroundLayers = defaultBgOrder.slice(0);
+
+          function setCurrentLayer(layerid) {
+            if (layerid) {
+              var layers = map.getLayers();
+              if (layerid == 'voidLayer') {
+                if (layers.getLength() > 0 &&
+                    layers.item(0).background === true) {
+                  layers.removeAt(0);
+                }
+              } else if (gaLayers.getLayer(layerid)) {
+                var layer = gaLayers.getOlLayerById(layerid);
+                layer.background = true;
+                layer.displayInLayerManager = false;
+                if (layers.item(0) && layers.item(0).background) {
+                  layers.setAt(0, layer);
+                } else {
+                  layers.insertAt(0, layer);
+                }
+              }
+              gaPermalink.updateParams({bgLayer: layerid});
+            }
+          }
+
+          scope.$on('gaLayersChange', function(event, data) {
+
+            // Determine the current background layer. Strategy:
+            //
+            // If this is the first gaLayersChange event we receive then
+            // we look at the permalink. If there's a bgLayer parameter
+            // in the permalink we use that as the initial background
+            // layer.
+            //
+            // If it's not the first gaLayersChange event, or if there's
+            // no bgLayer parameter in the permalink, then we use the
+            // first background layer of the background layers array.
+            //
+            // Specific use case when we go offline to online, in this use
+            // case we want to keep the current bg layer.
+
+            var bgLayer;
+            if (firstLayerChangeEvent) {
+              bgLayer = gaPermalink.getParams().bgLayer;
+              firstLayerChangeEvent = false;
+            }
+            if ((!bgLayer && !scope.currentLayer) ||
+              (scope.currentTopic && (scope.currentTopic != data.topicId))) {
+              bgLayer = gaLayers.getBackgroundLayers()[0].id;
+              scope.backgroundLayers = defaultBgOrder.slice(0);
+            }
+            if (bgLayer && !isOfflineToOnline) {
+              scope.currentLayer = bgLayer;
+            }
+            isOfflineToOnline = false;
+            scope.currentTopic = data.topicId;
+          });
+
+          scope.$on('gaPermalinkChange', function(event) {
+            scope.isBackgroundSelectorClosed = true;
+          });
+
+          scope.$watch('currentLayer', function(newVal, oldVal) {
+            if (oldVal !== newVal) {
+              setCurrentLayer(newVal);
+            }
+          });
+
+          scope.onClick = function(layerid) {
```

Comment:
It's usually better to name functions on what they do, not on what might trigger them. So maybe something like 'activateBackgroundLayer'.